worker_pool.go

```diff
@@ -0,0 +1,48 @@
+package wavelet
+
+import "sync"
+
+type WorkerPool struct {
+	bus    chan func()
+	stop   chan struct{}
+	wg     sync.WaitGroup
+	closed bool
+}
+
+func NewWorkerPool() *WorkerPool {
+	return &WorkerPool{
+		stop: make(chan struct{}),
+		bus:  make(chan func()),
+	}
+}
+
+func (wp *WorkerPool) Start(workers int) {
+	wp.wg.Add(workers)
+	for i := 0; i < workers; i++ {
+		go func() {
+			defer wp.wg.Done()
+			for {
+				select {
+				case job := <-wp.bus:
+					job()
+				case <-wp.stop:
+					return
+				}
+			}
+		}()
+	}
+}
+
+func (wp *WorkerPool) Queue(job func()) {
+	wp.bus <- job
```

Comment:
Can we add a check here to do something if the WorkerPool has been closed ?   Maybe `panic` or something ?

Reply:
i think i should close `bus` channel and then here will be panic because of that

Reply:
Perfect